src/PostgREST/App.hs

```diff
@@ -207,9 +174,11 @@ app dbstructure conf authenticator reqBody dbrole req =
         then do
           let call = B.Stmt "select " V.empty True <>
                 asJson (callProc qi $ fromMaybe M.empty (decode reqBody))
-          body :: Maybe (Identity Text) <- H.maybeEx call
+          bodyJson :: Maybe (Identity Value) <- H.maybeEx call
           return $ responseLBS status200 [jsonH]
-            (cs $ fromMaybe "[]" $ runIdentity <$> body)
+                (if hasPrefer "return=jwt"
```

Comment:
True, maybe we should use a different schema for functions that can be used to return JWT? 

Another approach is to use a convention about how the return value should look like. Let's say that there should be at least one column called role in the return (for the set role).

A third approach is to create a PG type called jwt_claims and say that functions returning that type are automatically converted to JWT (in this approach we can drop the use of the header).